File: BAEKJOON/알고리즘 기초 1/자료구조 1/C++/10828.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

#define endl '\n'

class stack
{
private:
	int arr[10000];
	int i = 0;

public:

	stack()
	{
		for (int j = 0; j < 10000; j++) {
			arr[i] = -1;
		}
		i = 0;
	}
	
	void push(int x)
	{
		arr[i++] = x;
		
	}

	int pop()
	{
		if (empty() == 1) {
			return -1;
		}
		i--;
		return arr[i];
	}

	int size()
	{
		return i;
	}

	int empty()
	{
		if (arr[i-1] == -1 || i == 0)
		{
			return 1;
		}
		
		return 0;
	}

	int top()
	{
		if (empty() == 1) return -1;

		return arr[i-1];
	}

};

int main(void)
{
	int n, tmp;
	cin >> n;
	string str;

	stack foo;
	for (int i = 0; i < n; i++) {
		cin >> str;
		if (str == "push") {
			cin >> tmp;
			foo.push(tmp);
		}
		else if (str == "pop") {
			cout << foo.pop() << endl;
		}
		else if (str == "size") {
			cout << foo.size() << endl;
		}
		else if (str == "empty") {
			cout << foo.empty() << endl;
		}
		else if (str == "top") {
			cout << foo.top() << endl;
		}
	}
	

	

	return 0;
}

File: BAEKJOON/알고리즘 기초 1/자료구조 1/python/10828.py
import sys

input = sys.stdin.readline

tmp = int(input())

stc = list()

for i in range(tmp):
    x = input().split()
    if x[0] == "push":
        stc.append(x[1])
    elif x[0] == "pop":
        if len(stc) == 0:
            print(-1)
        else: print(stc.pop())
    elif x[0] == "size":
        print(len(stc))
    elif x[0] == "empty":
        if len(stc) == 0:
            print(1)
        else: print(0)
    elif x[0] =="top":
        if len(stc) == 0:
            print(-1)
        else : print(stc[len(stc)-1])


        